src/ImageSharp/Formats/Tiff/Ifd/EntryReader.cs

```diff
@@ -3,317 +3,63 @@
-                    uint[] buf = new uint[count];
-                    for (int i = 0; i < buf.Length; i++)
-                    {
-                        buf[i] = this.stream.ReadUInt32();
-                    }
-
-                    return buf;
-                }
-
-                case ExifDataType.SignedLong:
-                {
-                    int[] buf = new int[count];
-                    for (int i = 0; i < buf.Length; i++)
-                    {
-                        buf[i] = this.stream.ReadInt32();
-                    }
-
-                    return buf;
-                }
-
-                case ExifDataType.Ascii:
-                {
-                    byte[] buf = this.stream.ReadBytes(count);
-
-                    if (buf[buf.Length - 1] != 0)
-                    {
-                        TiffThrowHelper.ThrowBadStringEntry();
-                    }
-
-                    return Encoding.UTF8.GetString(buf, 0, buf.Length - 1);
-                }
-
-                case ExifDataType.SingleFloat:
-                {
-                    float[] buf = new float[count];
-                    for (int i = 0; i < buf.Length; i++)
-                    {
-                        buf[i] = this.stream.ReadSingle();
-                    }
-
-                    return buf;
-                }
-
-                case ExifDataType.DoubleFloat:
-                {
-                    double[] buf = new double[count];
-                    for (int i = 0; i < buf.Length; i++)
-                    {
-                        buf[i] = this.stream.ReadDouble();
-                    }
-
-                    return buf;
-                }
-
-                case ExifDataType.Rational:
-                {
-                    var buf = new Rational[count];
-                    for (int i = 0; i < buf.Length; i++)
-                    {
-                        uint numerator = this.stream.ReadUInt32();
-                        uint denominator = this.stream.ReadUInt32();
-                        buf[i] = new Rational(numerator, denominator);
-                    }
-
-                    return buf;
-                }
-
-                case ExifDataType.SignedRational:
-                {
-                    var buf = new SignedRational[count];
-                    for (int i = 0; i < buf.Length; i++)
-                    {
-                        int numerator = this.stream.ReadInt32();
-                        int denominator = this.stream.ReadInt32();
-                        buf[i] = new SignedRational(numerator, denominator);
-                    }
-
-                    return buf;
-                }
-
-                case ExifDataType.Ifd:
-                {
-                    return this.stream.ReadUInt32();
-                }
-
-                default:
-                    return null;
-            }
-        }
+        protected override void RegisterExtLoader(uint offset, Action reader) => throw new NotImplementedException();
```

Comment:
I'm assuming this is a feature waiting to be implemented?

Reply:
The `HeaderReader` class used one time to read only header.